e2e/test/iothub/command/CommandE2ETests.cs

```diff
@@ -0,0 +1,238 @@
+                await digitalTwinClient.InvokeCommandAsync(
+                    deviceId,
+                    methodName,
+                    reqJson).ConfigureAwait(false);
+            }
+            else
+            {
+                response =
+                await digitalTwinClient.InvokeComponentCommandAsync(
+                    deviceId,
+                    componentName,
+                    methodName,
+                    reqJson).ConfigureAwait(false);
+            }
+
+
+            logger.Trace($"{nameof(DigitalTwinsSendCommandAndVerifyResponseAsync)}: Method status: {response.Response.StatusCode}.");
+            Assert.AreEqual(200, Shared.StatusCodes.OK, $"The expected response status should be 200 but was {response.Response.StatusCode}");
+            string payload = response.Body.Payload;
+            Assert.AreEqual(respJson, payload, $"The expected response payload should be {respJson} but was {payload}");
+
+            digitalTwinClient.Dispose();
+        }
+
+        public static async Task<Task> SubscribeAndUnsubscribeCommandAsync(DeviceClient deviceClient, string componentName, string methodName, MsTestLogger logger)
+        {
+            var methodCallReceived = new TaskCompletionSource<bool>();
+
+            await deviceClient
+                .SubscribeToCommandsAsync(
+                (request, context) =>
+                {
+                    if (string.IsNullOrEmpty(componentName))
+                    {
+                        logger.Trace($"{nameof(SubscribeAndUnsubscribeCommandAsync)}: DeviceClient method: {request.CommandName}.");
+                    }
+                    else
+                    {
+                        logger.Trace($"{nameof(SubscribeAndUnsubscribeCommandAsync)}: DeviceClient method: {request.ComponentName} {request.CommandName}.");
+                    }
+                    return Task.FromResult(new CommandResponse(new DeviceCommandResponse(), Shared.StatusCodes.OK));
+                },
+                null)
+                .ConfigureAwait(false);
+
+            await deviceClient.SubscribeToCommandsAsync(null, null).ConfigureAwait(false);
+
+            // Return the task that tells us we have received the callback.
+            return methodCallReceived.Task;
+        }
+
+        public static async Task<Task> SetDeviceReceiveCommandAsync(DeviceClient deviceClient, string componentName, string methodName, MsTestLogger logger)
+        {
+            var methodCallReceived = new TaskCompletionSource<bool>();
+
+            await deviceClient.SubscribeToCommandsAsync(
+                (request, context) =>
+                {
+                    logger.Trace($"{nameof(SetDeviceReceiveCommandAsync)}: DeviceClient method: {request.CommandName}.");
+
+                    try
+                    {
+                        var valueToTest = request.GetData<ServiceCommandRequestObject>();
+                        if (!string.IsNullOrEmpty(componentName))
```

Comment:
extreme nit: what do think about reversing the checks, since is it usually easier to evaluate the positive case than having to evaluate the expression and then reverse it

Reply:
Actually that's fine. I had my others this way